torch/csrc/distributed/autograd/context/dist_autograd_context.cpp

```diff
@@ -94,10 +96,23 @@ void DistAutogradContext::setGraphTask(
 
 void DistAutogradContext::addOutstandingRpc(
     const std::shared_ptr<rpc::FutureMessage>& futureMessage) {
+  futureMessage->addCallback(std::bind(
+      &DistAutogradContext::outStandingRpcCallback,
+      this,
+      std::placeholders::_1));
   std::lock_guard<std::mutex> guard(lock_);
   outStandingRpcs_.push_back(futureMessage);
 }
 
+void DistAutogradContext::outStandingRpcCallback(const rpc::Message& message) {
+  if (message.type() == rpc::MessageType::EXCEPTION) {
+    // If we have an error, let the local autograd engine know about it.
+    std::string err(message.payload().begin(), message.payload().end());
+    graphTask_->set_exception(
+        std::make_exception_ptr(std::runtime_error(err)), nullptr);
```

Comment:
It seems a bit weird that you copy into a `std::string`, and then pass a const reference to the string to the exception constructor which will do another copy